drivers/include/ds3231.h

```diff
@@ -101,6 +153,96 @@ int ds3231_get_time(const ds3231_t *dev, struct tm *time);
  */
 int ds3231_set_time(const ds3231_t *dev, const struct tm *time);
 
+/**
+ * @brief   Set alarm 1 of the device
+ *
+ * @param[in] dev       DS3231 device descriptor
+ * @param[in] time      target date and time
+ * @param[in] trigger   alarm 1 trigger type
+ *
+ * @return  0 on success
+ * @return  -EIO on I2C communication error
+ */
+int ds3231_set_alarm_1(const ds3231_t *dev, struct tm *time,
+                       ds3231_alm_1_mode_t trigger);
+
+/**
+ * @brief   Set alarm 2 of the device
+ *
+ * @param[in] dev       DS3231 device descriptor
+ * @param[in] time      target date and time
+ * @param[in] trigger   alarm 2 trigger type
+ *
+ * @return  0 on success
+ * @return  -EIO on I2C communication error
+ */
+int ds3231_set_alarm_2(const ds3231_t *dev, struct tm *time,
+                       ds3231_alm_2_mode_t trigger);
+
+/**
+ * @brief   Clear alarm 1 flag (A1F)
+ *
+ * @param[in] dev       DS3231 device descriptor
+ *
+ * @return  0 on success
+ * @return  -EIO on I2C communication error
+ */
+int ds3231_clear_alarm_1_flag(const ds3231_t *dev);
+
+/**
+ * @brief   Clear alarm 2 flag (A2F)
+ *
+ * @param[in] dev       DS3231 device descriptor
+ *
+ * @return  0 on success
+ * @return  -EIO on I2C communication error
+ */
+int ds3231_clear_alarm_2_flag(const ds3231_t *dev);
+
+/**
+ * @brief   Get the state of alarm 1 flag (A1F)
+ *
```

Comment:
Maybe add that this is not needed when `ds3231_await_alarm` is used, otherwise I see people use this function after `ds3231_await_alarm` and wonder why it returns `false`.